apps/gateway/src/middlewares/ai-model-factory.ts

```diff
@@ -0,0 +1,110 @@
+import { Elysia } from "elysia";
+
+import type { createDbClient } from "@hebo/database/client";
+import type {
+  ProviderConfig,
+  ProviderName,
+} from "@hebo/database/src/types/providers";
+import { dbClient } from "@hebo/shared-api/middlewares/db-client";
+import supportedModels from "@hebo/shared-data/json/supported-models";
+import type { Models } from "@hebo/shared-data/types/models";
+
+import { createProvider } from "./ai-models/providers";
+
+import type { EmbeddingModel, LanguageModel } from "ai";
+
+type ModelConfig = Models[number];
+
+const toProviderModelId = (
+  supportedModel: (typeof supportedModels)[number],
+  providerName: ProviderName,
+): string => {
+  const entry = supportedModel.providers.find(
+    (provider) => providerName in provider,
+  ) as Record<ProviderName, string>;
+  return entry[providerName];
+};
+
+const resolveProviderName = (
+  supportedModel: (typeof supportedModels)[number],
+): ProviderName => {
+  // Currently, we just pick the first provider for a model, in future this will be more sophisticated
+  const provider = supportedModel.providers[0];
+  return Object.keys(provider)[0] as ProviderName;
+};
+
+const getModelConfig = async (
```

Comment:
Keep this in the ModelConfigService, implemented as a class initialized with a dbContext in model-config.ts